samplesheets/tests/test_views_api_taskflow.py

```diff
@@ -144,6 +235,646 @@ def test_post_created(self):
+        self._assert_alert_count(CREATE_ALERT, self.user, 1)
+        self._assert_alert_count(CREATE_ALERT, self.user_delegate, 1)
+
+
+class TestIrodsRequestUpdateAPIView(TestIrodsRequestAPIViewBase):
+    """Tests for IrodsRequestUpdateAPIView"""
+
+    def test_update(self):
+        """Test updating a request"""
+        update_data = {'path': self.path, 'description': 'Updated'}
+
+        with self.login(self.user_contrib):
+            self.client.post(
+                reverse(
+                    'samplesheets:api_irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data,
+            )
+            self.assertEqual(IrodsDataRequest.objects.count(), 1)
+            obj = IrodsDataRequest.objects.first()
+            response = self.client.put(
+                reverse(
+                    'samplesheets:api_irods_request_update',
+                    kwargs={'irodsdatarequest': obj.sodar_uuid},
+                ),
+                update_data,
+            )
+            self.assertEqual(response.status_code, 200)
+
+        obj.refresh_from_db()
+        self.assertEqual(obj.description, 'Updated')
+        self.assertEqual(obj.path, self.path)
+
+    def test_update_invalid_data(self):
+        """Test updating a request with invalid data"""
+        update_data = {'path': '/doesnt/exist', 'description': 'Updated'}
+
+        with self.login(self.user_contrib):
+            self.client.post(
+                reverse(
+                    'samplesheets:api_irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data,
+            )
+            self.assertEqual(IrodsDataRequest.objects.count(), 1)
+            obj = IrodsDataRequest.objects.first()
+            response = self.client.put(
+                reverse(
+                    'samplesheets:api_irods_request_update',
+                    kwargs={'irodsdatarequest': obj.sodar_uuid},
+                ),
+                update_data,
+            )
+            self.assertEqual(response.status_code, 400)
+
+        obj.refresh_from_db()
+        self.assertEqual(obj.description, 'bla')
+        self.assertEqual(obj.path, self.path)
+
+
+class TestIrodsRequestDeleteAPIView(TestIrodsRequestAPIViewBase):
+    """Tests for IrodsRequestDeleteAPIView"""
+
+    def test_delete(self):
+        """Test deleting a request"""
+        with self.login(self.user_contrib):
+            self.client.post(
+                reverse(
+                    'samplesheets:api_irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data,
+            )
+            self.assertEqual(IrodsDataRequest.objects.count(), 1)
+            self._assert_alert_count(CREATE_ALERT, self.user, 1)
+            self._assert_alert_count(CREATE_ALERT, self.user_delegate, 1)
```

Comment:
In these deletion tests we probably aren't that interested in request creation alerts. Also, as a minor point, after `with self.login()..` you don't have to indent anything except for redirect asserts (those still require the user to be logged in).